### PR TITLE
#116 gui refactor

### PR DESCRIPTION
**Pull request by BluestoneDE**

### CODE REVIEW

File: src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java [hunk 1]
```diff
@@ -46,14 +46,34 @@ public RentalObjectBuilder setLivingSpace(Double livingSpace) {
...
+            throw new IllegalArgumentException("Wohnfl\u00e4che muss numerisch sein!");
+        }
+        this.livingSpace = Double.parseDouble(livingSpace);
+        return this;
+    }
+
     public RentalObjectBuilder setColdRentPerSquareMeter(Double coldRentPerSquareMeter) {
         if (Validator.isEmpty(coldRentPerSquareMeter)) {
-            throw new IllegalArgumentException("Kaltmiete/qm^2 darf nicht leer sein!");
+            throw new IllegalArgumentException("Kaltrente pro m\u00b2 darf nicht leer sein!");
```
Comment (ngoldack) [inline src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java:61]: wat ist denn Kaltrente

File: src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java [hunk 2]
```diff
@@ -46,14 +46,34 @@ public RentalObjectBuilder setLivingSpace(Double livingSpace) {
...
         }
         this.coldRentPerSquareMeter = coldRentPerSquareMeter;
         return this;
     }
 
+    public RentalObjectBuilder setColdRentPerSquareMeter(String coldRentPerSquareMeter) {
+        if (Validator.isEmpty(coldRentPerSquareMeter)) {
+            throw new IllegalArgumentException("Kaltrente pro m\u00b2 darf nicht leer sein!");
```
Comment (ngoldack) [inline src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java:69]: +1

File: src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java [hunk 3]
```diff
@@ -46,14 +46,34 @@ public RentalObjectBuilder setLivingSpace(Double livingSpace) {
...
+        } else if (Validator.isNotNumeric(coldRentPerSquareMeter)) {
+            throw new IllegalArgumentException("Kaltrente pro m\u00b2 muss numerisch sein!");
```
Comment (ngoldack) [inline src/main/java/de/hausbau/management/builder/RentalObjectBuilder.java:71]: +1

**Review by ngoldack — APPROVED** [review]

**Review by BluestoneDE — COMMENTED** [review]
@ngoldack WTF, you understand that this results in a new entry everytime we "update" something

**Review by ngoldack — APPROVED** [review]

**Review by ngoldack — APPROVED** [review]

**Pull request merged**